### ISSUE TITLE
[PROPOSAL] Standardize branching strategy

### ISSUE THREAD

**Issue by andrross** [opensearch-project/opensearch-oci-object-storage#27]
The OpenSearch project generally follows a strategy of tracking 3 releases in parallel (next major, next minor, and current). See [RELEASING.md](https://github.com/opensearch-project/.github/blob/51fb6fad37485f52fac241c3493d62575a03fcbe/RELEASING.md?plain=1#L23-L31) for more details. This repository currently only has a main branch that is targeting version 1.2.4. Should we update to match the OpenSearch branching strategy. If not, why not?

Assuming the answer is yes, then the steps are (I think?):

- Create branch `1.2`, which will be the current commit on `main`, supporting version 1.2.4
- Create branch `1.3` and implement support for version 1.3.8
- Create branch `2.x` and implement support for 2.6.0
- Implement support for 3.0.0 on `main`

**Comment by anasalkouz** [comment]
@dblock not sure if you have some context on this repository
Is any reason why it is not following OpenSearch branching strategy?
Why it's not part of the OpenSearch release bundle?

**Comment by wbeckler** [comment]
I propose that this entire repo be moved here: https://github.com/opensearch-project/OpenSearch/tree/main/plugins in a subdirectory called "repository-oci," just like there is a repository-azure and a repository-s3. If not, then why not create repos for those other cloud file stores?

**Comment by wbeckler** [comment]
I've opened an issue suggesting this: https://github.com/opensearch-project/OpenSearch/issues/6148

**Comment by samuel-oci** [comment]
Hi @anasalkouz @andrross @wbeckler , some context:
Originally it was created as a separate repo in order to allow the community time to decide how to address object storage repository plugins in the community. The community feedback at the time was captured here:
https://github.com/opensearch-project/OpenSearch/issues/1854

I agree with @wbeckler suggestion, I think that moving it into the core for now will probably make sure it remains consistent with both versioning strategy and will make routine maintenance and upgrade efforts easier.

**Comment by samuel-oci** [comment]
Regarding the branching strategy, if we want to take more time before moving it into the core, using a proper conventional branching strategy can't hurt.

**Comment by samuel-oci** [comment]
I created a 1.3 branch from the current commit on main. After merging https://github.com/opensearch-project/opensearch-oci-object-storage/pull/33 I will create a 2.4 branch.
Going forward will keep the above branching strategy after any version upgrade. Closing this for now while the suggestion to move to the monolith is discussed here: https://github.com/opensearch-project/OpenSearch/tree/main/plugins.
